tests/parametric/test_parametric_endpoints.py

```diff
@@ -0,0 +1,732 @@
+The results of these unit tests are reported to the feature parity dashboard.
+Parametric endpoints that are not tested in this file are not yet supported.
+Avoid using these endpoints in the parametric tests.
+"""
+import json
+import pytest
+
+from utils.parametric.spec.trace import find_trace
+from utils.parametric.spec.trace import find_span
+from utils.parametric.spec.trace import find_span_in_traces
+from utils.parametric.spec.trace import retrieve_span_links
+from utils.parametric.spec.trace import find_only_span
+from utils import irrelevant, scenarios, features, context
+from utils.parametric._library_client import Link
+
+
+@scenarios.parametric
+@features.parametric_endpoint_parity
+class Test_Parametric_DDSpan_Start:
+    def test_start_span(self, test_agent, test_library):
+        """
+        Validates the /trace/span/start API creates a new span:
+
+        Supported Parameters:
+        - span_id: int
+        - name: str
+        - service: Optional[str]
+        - resource: Optional[str]
+        - parent_id: Optional[Union[str,int]]
+        - type: Optional[str]
+        - tags: Optional[List[Tuple[str, str]]]
+
+        Supported Return Values:
+        - span_id: Union[int, str]
+        - trace_id: Union[int, str]
+        """
+        with test_library:
+            with test_library.start_span("parent") as s1:
+                pass
+
+            with test_library.start_span(
+                "child", "myservice", "myresource", s1.span_id, "web", tags=[("hello", "monkeys"), ("num", "1")],
+            ) as s2:
+                pass
+
+        traces = test_agent.wait_for_num_traces(1)
+        trace = find_trace(traces, s1.trace_id)
+        assert len(trace) == 2
+
+        parent_span = find_span(trace, s1.span_id)
+        assert parent_span["name"] == "parent"
+
+        child_span = find_span(trace, s2.span_id)
+        assert child_span["name"] == "child"
+        assert child_span["service"] == "myservice"
+        assert child_span["resource"] == "myresource"
+        # nodejs and dotnet libraries returns span and trace_ids as strings
+        assert child_span["parent_id"] == int(s1.span_id)
+        assert child_span["type"] == "web"
+        assert child_span["meta"]["hello"] == "monkeys"
+        assert child_span["meta"]["num"] == "1"
+
+
+@scenarios.parametric
+@features.parametric_endpoint_parity
+class Test_Parametric_DDSpan_Finish:
+    def test_span_finish(self, test_agent, test_library):
+        """
+        Validates that /traces/span/finish API finishes a span and sends it to the agent.
+
+        Supported Parameters:
+        - span_id: Union[int, str]
+        Supported Return Values:
+        """
+        with test_library:
+            # Avoids calling start_span_generator.__exit__() since this method manually finishes the span.
+            start_span_generator = test_library.start_span("span")
+            s1 = start_span_generator.__enter__()
```

Comment:
why use `__enter__` rather than with?

Reply:
test_library.start_span.__exit__() calls span finish, we only want to execute the logic before the span object is yielded